internal/agent/device/config/controller.go

```diff
@@ -45,6 +67,19 @@ func (c *Controller) Sync(desired *v1alpha1.RenderedDeviceSpec) error {
 	if err != nil {
 		return fmt.Errorf("writing ignition files failed: %w", err)
 	}
+	return nil
+}
 
+func (c *Controller) ensureHooks(hooks *[]v1alpha1.DeviceConfigHookSpec) error {
+	for i := range *hooks {
+		hook := (*hooks)[i]
```

Comment:
What if a hook is deleted? Where is it removed ?

Reply:
good catch this is not supported yet

Reply:
It seems to me if a hook which is based on a path does not appear anymore in the spec - it will continue to be fully functional.   Unless I am missing something.

Reply:
added basic tests and ensured watches are torn down.